apm-sniffer/apm-sdk-plugin/thrift-plugin/src/main/java/org/apache/skywalking/apm/plugin/thrift/wrapper/ServerInProtocolWrapper.java

```diff
@@ -81,6 +84,24 @@ public TField readFieldBegin() throws TException {
             }
             return readFieldBegin();
         }
+        if (field.type == TType.STOP) {
+            boolean haveCreatedSpan =
+                    (boolean) ContextManager.getRuntimeContext().get(HAVE_CREATED_SPAN);
```

Comment:
I think there is an NPE risk.
```suggestion
            Boolean haveCreatedSpan =
                    ContextManager.getRuntimeContext().get(HAVE_CREATED_SPAN);
```
And you should check `haveCreatedSpan != null && !haveCreatedSpan`, agree?

Reply:
If this context doesn't exist, then it could not change to `boolean`.

Reply:
I think I forgot to initialize, Otherwise haveCreatedSpan cannot be false.
I think Object cannot auto cast to Boolean.

Reply:
I don't say any object. I was saying null casting to Boolean rather than boolean.

Reply:
👍 , done